LibTest/async/MultiStreamController/addStream_A03_t01.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Future addStream(
+///   Stream<T> source, {
+///   bool? cancelOnError,
+/// })
+/// Receives events from source and puts them into this controller's stream.
+///
+/// Returns a future which completes when the source stream is done.
+///
+/// Events must not be added directly to this controller using `add`,
+/// `addError`, `close` or `addStream`, until the returned future is complete.
+///
+/// Data and error events are forwarded to this controller's stream. A done
+/// event on the source will end the addStream operation and complete the
+/// returned future.
+///
+/// If `cancelOnError` is `true`, only the first error on `source` is forwarded
+/// to the controller's stream, and the `addStream` ends after this. If
+/// `cancelOnError` is `false`, all errors are forwarded and only a done event
+/// will end the `addStream`. If `cancelOnError` is omitted or null, it defaults
+/// to `false`.
+///
+/// @description Checks that if `cancelOnError` is `false` all errors are
+/// forwarded and only a done event ends the `addStream`.
+/// @author sgrekhov22@gmail.com
+
+import "dart:async";
+import "../../../Utils/expect.dart";
+
+main() {
+  asyncStart();
+  var source = StreamController<int>();
+  source.add(1);
+  source.add(2);
+  source.addError(-1);
+  source.add(3);
+  source.addError(-2);
+  source.close();
+
+  var stream = Stream<int>.multi((controller) {
+    controller.addStream(source.stream, cancelOnError: false).then((_) {
+      controller.close();
+    });
+  });
+  listen(stream);
+}
+
+void listen(Stream<int> stream) {
+  int eventsCounter = 1;
```

Comment:
Wouldn't it still work if you used `eventsCounter = 0` and then `++eventsCounter` in line 55? Seems slightly simpler to understand.

(Also, just as a side remark, `++x` is simpler than `x++` because the former is a side-effect and a variable-read operation, but the latter is more like "cache the old value in a tmp variable", then the side effect, then "evaluate to the tmp variable". So I prefer `++x` unless there is a specific need for `x++`.)

Reply:
Makes sense, thank you! Updated.